ubccr/xdmod
# Making sure to clear a users organizations before adding

## Description
the `XDUser::setOrganizations` only removes entries that correspond to the
`$aclName` provided. Which only accounts for creating / updating a
role / center not for removing one. ( If it's not there then it can't be
removed ). Previously this function was called regardless of whether or not a user had the center [director|staff] role ( if they didn't have it then it would be called in much the same manner as we have added here ).

Addresses: https://app.asana.com/0/558460821012916/558487556596711

## Motivation and Context
Consistency is good

## Tests performed
Manual Tests:
**Test 1:**
- Login as user with Center Director
- Click 'My Profile'
- Click 'Role Delegation'
- Select User from Dropdown
- Ensure that a button with the text 'Upgrade Center Staff Member to Center Director' is present
- Click the button
- Ensure that the following message is displayed: 
```
<User Formal Name> [<Center>]
has been upgraded to Center Director
(promoted by <Currently Logged In User Formal Name>)
```
- And that there is a button below the message with the text: `Revoke Center Director Privileges`.
- Click the Revoke button
- Ensure that the 'Upgrade Center Staff Member to Center Director' is back

## Types of changes


## Checklist:
